src/node/transaction.cpp

```diff
@@ -36,10 +36,19 @@ TransactionError BroadcastTransaction(NodeContext& node, const CTransactionRef t
         if (!existingCoin.IsSpent()) return TransactionError::ALREADY_IN_CHAIN;
     }
     if (!node.mempool->exists(hashTx)) {
-        // Transaction is not already in the mempool. Submit it.
+        // Transaction is not already in the mempool.
         TxValidationState state;
+        CAmount fee;
+        bool test_accepted = AcceptToMemoryPool(*node.mempool, state, std::move(tx),
+                nullptr /* plTxnReplaced */, false /* bypass_limits */, /* test_accept */ true, &fee);
+        if (test_accepted) {
+            if (max_tx_fee && fee > max_tx_fee) {
+                return TransactionError::MAX_FEE_EXCEEDED;
+            }
+        }
+        // Transaction fee is acceptable. Submit the transaction.
```

Comment:
Instead of calling ATMP twice, can we trust the `test_accepted` value from the first call and skip redundant checks in the second call? Or maybe to put it another way, if all we need is the fee here, can we compute that quickly and test it against `max_tx_fee` before checking anything else about the tx? I think if that's easy enough to do, you don't have to add an argument to ATMP. But I guess you'd still have to remove `nAbsurdFee` everywhere it's called.